contractcourt/htlc_outgoing_contest_resolver_test.go

```diff
@@ -0,0 +1,158 @@
+package contractcourt
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet"
+)
+
+const (
+	outgoingContestHtlcExpiry = 110
+)
+
+// TestHtlcOutgoingResolverTimeout tests resolution of an offered htlc that
+// timed out.
+func TestHtlcOutgoingResolverTimeout(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	ctx := newOutgoingResolverTestContext(t)
+	ctx.resolve()
+	ctx.notifyEpoch(outgoingContestHtlcExpiry - 1)
+	ctx.waitForResult(true)
+}
+
+// TestHtlcOutgoingResolverRemoteClaim tests resolution of an offered htlc that
+// is claimed by the remote party.
+func TestHtlcOutgoingResolverRemoteClaim(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	ctx := newOutgoingResolverTestContext(t)
+	ctx.resolve()
+	preimage := lntypes.Preimage{}
+	ctx.notifier.spendChan <- &chainntnfs.SpendDetail{
+		SpendingTx: &wire.MsgTx{
+			TxIn: []*wire.TxIn{
+				{
+					Witness: [][]byte{
+						{0}, {1}, {2}, preimage[:],
+					},
+				},
+			},
+		},
+	}
+	<-ctx.preimageDB.newPreimages
```

Comment:
It's not immediately apparent to me why we need to block on this? I see the `ChainArbitratorConfig` shares a `preimageDB` with `ctx`, could you add a comment which explains the interaction here so it's clearer why we need to unblock it?

Reply:
comment added. we are asserting the the resolver is saving the preimage that it extracted from the chain tx